src/LinearSolvers/preconditioners.jl

```diff
@@ -55,18 +79,28 @@ mutable struct BlockJacobiPreconditioner <: AbstractPreconditioner
         for (i,v) in enumerate(part)
             push!(partitions[v], i)
         end
-        Js = Vector{Matrix{Float64}}(undef, npart)
-        nJs = maximum(length.(partitions))
-        id = Matrix{Float64}(I, nJs, nJs)
+        lpartitions = VI(undef, npart)
+        for i in 1:npart
+            lpartitions[i] = length(partitions[i])
+        end
+        blocksize = maximum(length.(partitions))
+        blocks = AT(undef, blocksize, blocksize, npart)
```

Comment:
I would be careful there. E.g. 
```julia
a = rand(100)
VT = CuVector{Float64}
@time VT(a);
  0.000097 seconds (10 allocations: 240 bytes)

@time convert(VT, a);
  0.000084 seconds (7 allocations: 192 bytes)
```
It's usually better to use `convert` rather than `VT(...)`.

Reply:
Also, if the type is already set to the destination:
```julia
VT = Vector{Float64}
@time convert(VT, a);
  0.000005 seconds
@time VT(a);
  0.000006 seconds (1 allocation: 896 bytes)
```
One version is allocating, the other not

Reply:
I tried to dig into the source code, because I don't understand the difference. First, if I rerun the code I get this:
```julia

julia> @time VT(a)
  0.000088 seconds (9 allocations: 224 bytes)

julia> @time convert(VT, a)
  0.000082 seconds (9 allocations: 224 bytes)
```
But that's Julia 1.5.3.